### ISSUE TITLE
The Twitter parser is broken

### ISSUE THREAD

**Issue by marvinborner** [Jugendhackt/JugendhacktApp#43]
The feed on alpaca.ml shows the error message `Invalid date`. I haven't investigated it yet but it seems like our parser is broken.

Edit: The backend behind alpaca.ml does not log anything weird and the request returns `[]`. So it really seems like Twitter changed something.

**Comment by LarsVomMars** [comment]
https://github.com/desmondmorris/node-twitter

**Comment by marvinborner** [comment]
We would need a token for this and the whole purpose of a self-written scraper was that we can parse 
without a token :)

**Comment by LarsVomMars** [comment]
Might end up being better tho :)

**Comment by marvinborner** [comment]
We could also try to fix the issue instead of throwing some nodejs libs into the project

**Comment by LarsVomMars** [comment]
Select all `article` tags from the page

**Comment by marvinborner** [comment]
While this seems to select the correct item, the content div doesn't exist anymore and is split up into several tags. We may have to loop over them.

**Comment by LarsVomMars** [comment]
Yeah, ik

**Comment by Mondanzo** [comment]
Leave me out of your parser mess. I make front-end not back-end D:

**Comment by LarsVomMars** [comment]
`a.children[0].children[1].children[1].children[1] ` for a as the article would be the contents

**Comment by marvinborner** [comment]
It seems like the parser isn't even the problem. The JSDOM call doesn't return the correct DOM as all the tweets are missing (`console.log(dom.window.document.body.innerHTML`).
Or I'm just dumb.

**Comment by Kurty00** [comment]
It is not recommended to scrape from Twitter since it states in the TOS that "scraping the Services without the prior consent of Twitter is expressly prohibited" (§4 Twitter TOS)

**Comment by marvinborner** [comment]
Yeah, we know. This was a temporary solution which seemed quite acceptable for a hackathon application. We could switch to the official API if someone wants to invest the time.

**Comment by Mondanzo** [comment]
Let's just drop Twitter support and add Mastodon support.

**Comment by marvinborner** [comment]
That's also a possibility :)
